### PR TITLE
Wrap client and server into a Service

### PR DESCRIPTION
**Pull request by dennwc**

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -83,10 +85,22 @@ func runService(c *cli.Context) error {
 	killChan := make(chan os.Signal, 1)
 	signal.Notify(killChan, syscall.SIGINT)
 
-	svc := service.NewService(conf, psrpcClient, bus)
+	ua, err := sipgo.NewUA(
```
Comment (Sean-Der) [inline cmd/server/main.go]: @dennwc Why not put this inside the `SIPService` ?

Comment (dennwc) [reply cmd/server/main.go]: Good point! I'll make it init both SIP client and the server in the service constructor.

File: cmd/server/main.go [hunk 2]
```diff
@@ -83,26 +81,14 @@ func runService(c *cli.Context) error {
 	killChan := make(chan os.Signal, 1)
 	signal.Notify(killChan, syscall.SIGINT)
 
-	svc := service.NewService(conf, psrpcClient, bus)
-
-	sipSrv := sip.NewServer(conf, svc.HandleTrunkAuthentication, svc.HandleDispatchRules)
```
Comment (Sean-Der) [inline cmd/server/main.go:94]: @dennwc the split of ‘service’ and ‘sip service’ comes from ingress/egress. I think it’s best to keep it that way so things feel similar

Comment (dennwc) [reply cmd/server/main.go:94]: Updated. Is this what you had in mind?

**Pull request merged**